package fesql.type;
option cc_generic_services = false;
option java_package = "com._4paradigm.fesql.type";
option java_outer_classname = "Type";

enum Type {
    kBool = 0;
    kInt16 = 1;
    kUInt16 = 2;
    kInt32 = 3;
    kUInt32 = 4;
    kInt64 = 5;
    kUInt64 = 6;
    kFloat = 7;
    kDouble = 8;
    kString = 9;
    kDate = 10;
    kTimestamp = 11;
}

message ColumnDef {
    optional string name = 1;
    optional Type type = 2;
    optional uint16 offset = 3;
}

message User {
    optional string name = 1;
    optional string token = 2;
    optional uint64 id = 3;
    optional uint64 ctime = 4;
}

message Group {
    optional string name = 1;
    optional uint64 id = 2;
    optional uint64 ctime = 3;
    optional uint64 owner = 4;
}

message CataLog {
    optional string name = 1;
    optional uint64 id = 2;
    optional uint64 ctime = 3;
    optional uint64 gid = 4;
    optional uint64 utime = 5;
}

message TableDef {
    repeated ColumnDef columns = 1;
    optional string name = 2;
    optional uint64 id = 3;
    optional uint64 catalog_id = 4;
    optional uint64 ctime = 5;
    optional uint64 utime = 6;
}
